rfc/java11.md

```diff
@@ -0,0 +1,171 @@
+*   Debating which Java version we should upgrade to; we take it as a given that
+    only LTS releases are acceptable for most organizations, especially for
+    running critical infrastructure like Spinnaker
+*   Documenting and solving technical issues that present themselves in the
+    migration; those will be handled separately, but this document concerns
+    itself primarily with the larger processes and timeline
+*   Detailing the required release system changes; since the releases are still
+    entirely maintained by Google, these changes aren't particularly relevant to
+    the wider community
+
+### JRE vs. JDK
+
+There are two components of Java of interest to this RFC. The Java Runtime
+Environment (JRE) is the `java` binary and associated libraries used to _run_
+Spinnaker services. The Java Development Kit (JDK) is the `javac` binary and
+associated libraries used to _compile_ Spinnaker.
+
+Once we upgrade to the Java 11 JDK, Spinnaker will no longer be able to run in a
+Java 8 JRE. So the first (more complicated) step is to get everyone migrated to
+a Java 11 JRE. Only after that is done can we switch to the Java 11 JDK.
+
+The Java 11 JRE has some fairly large changes, most notably the deprecation of
+the long-standing CMS garbage collector. Large organizations with highly-tuned
+JVMs have shown hesitancy to upgrade until the new runtime has undergone
+extensive testing. (Both Google and Netflix are in this category.)
+
+For official Spinnaker containers, we control the JRE used to run Spinnaker. For
+those releases, this proposal takes a relatively slow ramp to the Java 11 JRE
+and provides an escape hatch back to Java 8 for users who experience problems.
+
+There are also some unknown number of customers who aren't using the Spinnaker-
+provided containers. We will need to do some outreach to make sure these users
+convert to a Java 11 JRE too.
+
+## Timeline
+
+### 1.18 (early January)
+
+For our container builds, we'll first move just two (of approximately ten)
+services to the new JRE. Thee two services will be `igor` and `orca`. By
+default, anyone who uses Halyard to deploy these containers will get the Java 11
+JRE for these services. Users who experience problems or aren't ready to use
+Java 11 can use the `java8` variant by specifying `imageVariant: slim-java8` (or
+`ubuntu-java8`) in their Halyard config file.
+
+The release notes will prominently mention this fact. It will also be mentioned
+in-channel in the release announcement to `#spinnaker-releases` and in the
+release announcement email. We will strongly encourage people who opt for the
+`java8` variants to contact `sig-platform@spinnaker.io` or `#sig-platform` to
+discuss their concerns, which will allow us to adjust the timeline as needed.
+
+Separately, postings to `#dev` will encourage developers to start running their
+development environments under Java 11. Explanations of how to do so will be
+posted to the channel.
```

Comment:
s/channel/spinnaker.io website. Channel instructions scroll off the screen pretty easily.

Reply:
Good call, thanks I added this:

> The [Spinnaker Developer Guide](https://www.spinnaker.io/guides/developer/getting-set-up/) will also be updated to strongly recommend using Java 11.